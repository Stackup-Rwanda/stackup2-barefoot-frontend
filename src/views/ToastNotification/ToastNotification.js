import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
  },
}));

const ToastNotification = ({ error }) => {
  const classes = useStyles();
  const [open] = useState(true);

  return (
    <div className={classes.root}>
      <Snackbar open={open} autoHideDuration={6000}>
        <MuiAlert elevation={6} variant="filled" severity="error">
          {error}
        </MuiAlert>
      </Snackbar>
    </div>
  );
};

ToastNotification.propTypes = {
  error: PropTypes.string.isRequired,
};

export default ToastNotification;
