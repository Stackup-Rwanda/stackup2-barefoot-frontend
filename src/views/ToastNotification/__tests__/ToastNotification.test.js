/* eslint-disable no-undef */
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import { cleanup } from '@testing-library/react';
import ToastNotification from '../ToastNotification';

const error = 'Some error message';

describe('<ToastNotification />', () => {
  afterEach(() => cleanup);

  it('Should match the ToastNotification snapshot', () => {
    const component = createMount()(
      <ToastNotification error={error} />,
    );
    expect(component.html).toMatchSnapshot();
  });
});

// /* eslint-disable no-undef */
// import React from 'react';
// import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
// import { cleanup } from '@testing-library/react';
// import ToastNotification from '../ToastNotification';

// const error = 'Some error message';

// describe('<ToastNotification/>', () => {
//   afterEach(() => cleanup);

//   it('Should render the SectionHeader component', () => {
//     const tree = shallow(<ToastNotification error={error} />);
//     expect(tree.find('Snackbar').exists()).toBe(true);
//     expect(tree.find('MuiAlert').exists()).toBe(true);
//   });
// });
