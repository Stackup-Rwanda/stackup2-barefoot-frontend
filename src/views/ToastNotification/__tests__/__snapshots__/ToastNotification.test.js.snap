// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastNotification /> Should match the ToastNotification snapshot 1`] = `[Function]`;
